### ISSUE TITLE
Cut & Paste only pastes first paragraph

### ISSUE THREAD

**Issue by xpurichan** [Automattic/wp-calypso#80757]
### Quick summary

When we cut multiple paragraphs of text and then go to paste it, we are experiencing that only the first paragraph pastes within the WordPress editor. Adding that if we cut and paste in to another program then all paragraphs are pasted. 

Using the undo arrow within the WordPress editor then either fails to restore the missing paragraphs (if the Gutenberg plugin is _de_activated), or causes the editor to fail with a "The editor has encountered an unexpected error." Error text copied below:

```
Error: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at qr (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:75172)
    at $r (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:74857)
    at qr (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:75614)
    at $r (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:74857)
    at qr (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:75614)
    at $r (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:74857)
    at qr (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:75806)
    at $r (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:74857)
    at qr (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:75614)
    at $r (https://lillylearns.com/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:1:74857)
```

Noting that in either case if we do not save the changes and refresh the editor then the missing text is then restored.

### Steps to reproduce

**Considerations**
The issue has been observed on self-hosted sites and on Atomic with all plugins deactivated except for Jetpack (which was disconnected) and Akismet. **I could not reproduce on a simple site.** The behavior is sometimes inconsistent if you use keyboard shortcuts to cut & paste or have spotlight enabled, so focus on using right(command) click and having spotlight disabled.

**A video recording of the issue:**
https://www.loom.com/share/fdaef4278f6a4c969ca03fb328ede169?sid=18aa5b05-283a-47bd-9bd5-7d9adc7def0c

1. On an Atomic or self-hosted site open the post or page editor
2. Highlight the text across at least two blocks - testing was performed with two paragraph blocks
3. Right-click and _cut_ the text
4. Right-click and _paste_ the text
5. Observe that only the first paragraph block was pasted back in to the editor, the second and subsequent blocks are now missing.
6. Using the undo/back arrow within the editor either causes an unexpected error if Gutenberg is active, or simply fails to restore the missing text if Gutenberg is not activated.

### What you expected to happen

Pasting text that has been cut should result in all text being restored.

### What actually happened

Cutting and then pasting text only returns the first paragraph, subsequent paragraphs are not pasted.

### Impact

All

### Available workarounds?

Yes, mentioned above:

>_The behavior is sometimes inconsistent if you use keyboard shortcuts to cut & paste or have spotlight enabled_

### Platform (Simple and/or Atomic)

Atomic, Self-hosted

**Comment by annezazu** [comment]
Noting that I have escalated this to the Gutenberg github repo after replicating with 6.3: https://github.com/WordPress/gutenberg/issues/53774 We're looking at doing a 6.3.1 soon enough and hopefully a fix can be included there.

**Comment by liviopv** [comment]
Just documenting that this affects CUT only. COPY is working fine.

https://github.com/Automattic/wp-calypso/assets/14153300/6654ca53-a484-4e20-ad96-932c28fa0209

And it only happens when using the mouse. Cmmd/Ctrl + X works fine.

**Comment by cuemarie** [comment]
📌 FINDINGS/SCREENSHOTS/VIDEO
- As shared above, this is happening upstream and a fix is in the works!  See: https://github.com/WordPress/gutenberg/issues/53774

📌 ACTIONS
- Assigned to Other

**Comment by xpurichan** [comment]
This issue was resolved in WP 6.3.1 - https://github.com/WordPress/gutenberg/issues/53774#issuecomment-1682933457
